src/cmd/build.js

```diff
@@ -60,8 +65,16 @@ export default function build(
   const rebuildAsNeeded = asNeeded; // alias for `build --as-needed`
   log.info(`Building web extension from ${sourceDir}`);
 
+  const fileExtension = extension; // alias for `build --extension <ext>`
+  let allowedExtension = /^[a-zA-Z0-9_-]+$/;
+  if (!allowedExtension.test(fileExtension)) {
+    throw Error('Only alphanumeric, -, _ characters are ' +
```

Comment:
this should throw [WebExtError](https://github.com/mozilla/web-ext/blob/master/src/errors.js#L8).